src/passes/pass.cpp

```diff
@@ -402,15 +413,23 @@ void PassRunner::addDefaultFunctionOptimizationPasses() {
   // simplify-locals opens opportunities for optimizations
   add("remove-unused-brs");
   // if we are willing to work hard, also optimize copies before coalescing
-  if (options.optimizeLevel >= 3 || options.shrinkLevel >= 2) {
+  // FIXME DWARF updating does not handle local changes yet.
+  if (!preserveDWARF &&
+      (options.optimizeLevel >= 3 || options.shrinkLevel >= 2)) {
     add("merge-locals"); // very slow on e.g. sqlite
   }
-  add("coalesce-locals");
+  // FIXME DWARF updating does not handle local changes yet.
+  if (!preserveDWARF) {
+    add("coalesce-locals");
+  }
   add("simplify-locals");
```

Comment:
does simplify-locals need to be excluded too?

Reply:
No, it just removes unneeded sets, it doesn't rename them.